cloud/amazon/dynamodb_table.py

```diff
@@ -222,16 +251,43 @@ def dynamo_table_exists(table):
             raise e
 
 
-def update_dynamo_table(table, throughput=None, check_mode=False):
+def update_dynamo_table(table, throughput=None, check_mode=False, global_indexes=None):
     table.describe()  # populate table details
-
+    throughput_changed = False
+    global_indexes_changed = False
     if has_throughput_changed(table, throughput):
         if not check_mode:
-            return table.update(throughput=throughput)
+            throughput_changed = table.update(throughput=throughput)
         else:
-            return True
+            throughput_changed = True
 
-    return False
+    removed_indexes, added_indexes, index_throughput_changes = get_changed_global_indexes(table, global_indexes)
+    if removed_indexes:
+        if not check_mode:
+            for name, index in removed_indexes.iteritems():
+                global_indexes_changed = table.delete_global_secondary_index(name) or global_indexes_changed
+        else:
+            global_indexes_changed = True
+
+    if added_indexes:
+        if not check_mode:
+            for name, index in added_indexes.iteritems():
+                # raise Exception(str(index.schema()))
+                global_indexes_changed = table.create_global_secondary_index(global_index=index) or global_indexes_changed
+        else:
+            global_indexes_changed = True
+
+    if index_throughput_changes:
+        if not check_mode:
+            # todo: remove try once boto has https://github.com/boto/boto/pull/3447 fixed
```

Comment:
I left todos for future work based on working around a boto bug. Let me know if I should remove them